src/machines/java/com/enderio/machines/client/gui/screen/MachineScreen.java

```diff
@@ -0,0 +1,34 @@
+package com.enderio.machines.client.gui.screen;
+
+import com.enderio.core.client.gui.screen.EIOScreen;
+import com.enderio.machines.common.menu.GhostMachineSlot;
+import com.enderio.machines.common.menu.MachineMenu;
+import com.enderio.machines.common.menu.PreviewMachineSlot;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.inventory.Slot;
+
+public abstract class MachineScreen<T extends MachineMenu> extends EIOScreen<T> {
+    protected MachineScreen(T pMenu, Inventory pPlayerInventory, Component pTitle) {
+        super(pMenu, pPlayerInventory, pTitle);
+    }
+
+    protected MachineScreen(T pMenu, Inventory pPlayerInventory, Component pTitle, boolean renderLabels) {
+        super(pMenu, pPlayerInventory, pTitle, renderLabels);
+    }
+
+    @Override
+    public void renderSlot(GuiGraphics guiGraphics, Slot slot) {
+        super.renderSlot(guiGraphics, slot);
+
+        if (slot instanceof GhostMachineSlot || slot instanceof PreviewMachineSlot) {
+            if (slot.hasItem()) {
+                guiGraphics.pose().pushPose();
+                guiGraphics.pose().translate(0.0F, 0.0F, 300F);
+                guiGraphics.fill(slot.x, slot.y, slot.x + 16, slot.y + 16, -2130706433);
```

Comment:
Very minor - could we have a named constant for that colour so it's easier to understand in future? Cheers!

Reply:
since we dont have a constants file yet I have created a public static field in the same class